kubeflow/trainer/rhai/utils.py

```diff
@@ -30,8 +36,99 @@ def get_trainer_cr_from_rhai_trainer(
         )
 
     else:
-        _raise_unknown_rhai_trainer(trainer)
+        raise ValueError(f"Unknown trainer {trainer}.")
+
+
+def parse_output_dir_uri(output_dir: Optional[str]) -> tuple[Optional[str], Optional[dict]]:
+    """
+    Parse output_dir URI and return resolved path and volume mount specs.
+    """
+    if not output_dir:
+        return None, None
+
+    if output_dir.startswith(PVC_URI_SCHEME):
+        # Parse URI and split into PVC name and path
+        uri_path = output_dir[len(PVC_URI_SCHEME) :]
+        parts = uri_path.split("/", 1)
+        pvc_name = parts[0]
+        checkpoint_path = parts[1] if len(parts) > 1 else ""
+
+        # SDK mounts PVC at a standard location
+        mount_path = CHECKPOINT_MOUNT_PATH
+        resolved_path = f"{mount_path}/{checkpoint_path}" if checkpoint_path else mount_path
+
+        # Build Kubernetes volume and volumeMount specs
+        volume_spec = {
+            "name": CHECKPOINT_VOLUME_NAME,
+            "persistentVolumeClaim": {"claimName": pvc_name},
+        }
+        volume_mount_spec = {"name": CHECKPOINT_VOLUME_NAME, "mountPath": mount_path}
+
+        return resolved_path, {"volume": volume_spec, "volumeMount": volume_mount_spec}
+
+    return output_dir, None
+
+
+def apply_output_dir_uri_to_pod_overrides(
+    output_dir: str,
+    pod_template_overrides: Optional[list],
+) -> tuple[str, list]:
+    """
+    Process output_dir URI and apply PVC mounting to pod template overrides.
+    """
+    resolved_output_dir, volume_mount_specs = parse_output_dir_uri(output_dir)
+
+    # If no PVC mounting needed return none
+    if volume_mount_specs is None:
+        return resolved_output_dir, pod_template_overrides
```

Comment:
At this point, there might be case where due to early return the function returns [out_dir, None] which doesn't match with return type tuple[str,list] or can raise valueError.. WDYT?
```suggestion
        return resolved_output_dir, pod_template_overrides or []
```